core/src/main/java/io/kestra/core/server/ServiceRegistry.java

```diff
@@ -60,6 +60,29 @@ public List<LocalServiceState> all() {
         return new ArrayList<>(services.values());
     }
 
+    /**
+     * Waits for a given service to be in a given state if registered.
+     *
+     * @param type            The service type
+     * @param state           The expected state.
+     * @param maxWaitDuration The max wait duration.
+     * @return {@code true} if the service is in the expected state. Otherwise {@code false}.
+     */
+    public boolean waitForServiceInState(final Service.ServiceType type,
```

Comment:
Couldn't see any usage :thinking:

Reply:
This method has been added in another commit and used in EE testing. Initially, the idea was to use it directly in the `RunnerUtils`  to provide more stability to tests. But, the use of the `RunnerUtils` class was not very clear. After discussing this with Loic, it seems to be a good idea to refactor the `RunnerUtils` to ensure it is only used for testing purpose. For the moment, the class still contains the deprecated methods to not break any plugins in 0.16.0. In 0.17.0 we should perform a removal of all deprecated methods to cleanup everythings.